lib/matplotlib/figure.py

```diff
@@ -1528,635 +965,1865 @@ def subplots(self, nrows=1, ncols=1, sharex=False, sharey=False,
+
+        frameon : bool, default: :rc:`figure.frameon`
+            If ``False``, suppress drawing the figure background patch.
+        """
+        super().__init__()
+        if facecolor is None:
+            facecolor = mpl.rcParams['figure.facecolor']
+        if edgecolor is None:
+            edgecolor = mpl.rcParams['figure.edgecolor']
+        if frameon is None:
+            frameon = mpl.rcParams['figure.frameon']
+
+        self._subplotspec = subplotspec
+        self._parent = parent
+        self.figure = parent.figure
+        # subfigures use the parent axstack
+        self._axstack = parent._axstack
+        self.subplotpars = parent.subplotpars
+        self.dpi_scale_trans = parent.dpi_scale_trans
+        self._axobservers = parent._axobservers
+        self.dpi = parent.dpi
+        self.canvas = parent.canvas
+        self.transFigure = parent.transFigure
+        self.bbox_relative = None
+        self._redo_transform_rel_fig()
+        self.figbbox = self._parent.figbbox
+        self.bbox = TransformedBbox(self.bbox_relative,
+                                    self._parent.transSubfigure)
+        self.transSubfigure = BboxTransformTo(self.bbox)
+
+        self.patch = Rectangle(
+            xy=(0, 0), width=1, height=1, visible=frameon,
+            facecolor=facecolor, edgecolor=edgecolor, linewidth=linewidth,
+            # Don't let the figure patch influence bbox calculation.
+            in_layout=False, transform=self.transSubfigure)
+        self.patch.set_antialiased(False)
+
+        if parent._layoutgrid is not None:
+            self.init_layoutgrid()
+
+    def _redo_transform_rel_fig(self, bbox=None):
+        """
+        Make the transSubfigure bbox relative to Figure transform.
+
+        Parameters
+        ----------
+        bbox : bbox or None
+            If not None, then the bbox is used for relative bounding box.
+            Otherwise it is calculated from the subplotspec.
+        """
+
+        if bbox is not None:
+            self.bbox_relative.p0 = bbox.p0
+            self.bbox_relative.p1 = bbox.p1
+            return
+
+        gs = self._subplotspec.get_gridspec()
+        # need to figure out *where* this subplotspec is.
+        wr = gs.get_width_ratios()
+        hr = gs.get_height_ratios()
+        nrows, ncols = gs.get_geometry()
+        if wr is None:
+            wr = np.ones(ncols)
+        else:
+            wr = np.array(wr)
+        if hr is None:
+            hr = np.ones(nrows)
+        else:
+            hr = np.array(hr)
+        widthf = np.sum(wr[self._subplotspec.colspan]) / np.sum(wr)
+        heightf = np.sum(hr[self._subplotspec.rowspan]) / np.sum(hr)
+
+        x0 = 0
+        if self._subplotspec.colspan[0] > 0:
+            x0 += np.sum(wr[self._subplotspec.colspan[0] - 1]) / np.sum(wr)
+
+        y0 = 0
+        if self._subplotspec.rowspan[-1] < nrows - 1:
+            y0 += 1 - np.sum(hr[self._subplotspec.rowspan[-1]-1]) / np.sum(hr)
```

Comment:
Are these not (I'm not sure about the `-1`s)?
```suggestion
        x0 = 0
        if not self._subplotspec.is_first_col():
            x0 += np.sum(wr[self._subplotspec.colspan.start - 1]) / np.sum(wr)

        y0 = 0
        if not self._subplotspec.is_last_row():
            y0 += 1 - np.sum(hr[self._subplotspec.rowspan.end - 1]) / np.sum(hr)
```